[ryanjansen] iP

ryanjansen opened the pull request:
### Duke

Duke is a simple, easy to use task manager for those who can't remember jack.

**Features**
- Add tasks
- Delete tasks
- Mark tasks as done

**How to add a task**
1. Type "todo" followed by the title to add a todo
2. Type "event" followed by the title, from date and to date to add a event
3. Type "deadline" followed by the title, date, to add deadline

Here's a demo of checking a task:

- [ ] A task
- [ ] Another task

Here's an example of the code:

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

Commit:
Add Gradle support

ryanjansen committed:
Implemented Level-1

ryanjansen committed:
Implemented Level 2

ryanjansen committed:
Implemented Level 3

ryanjansen committed:
Implemented Level 4

ryanjansen committed:
Implemented Text UI Testing

ryanjansen committed:
Implemented Level 5

ryanjansen committed:
Implemented Level 6

ryanjansen committed:
Implemented Level 7

ryanjansen committed:
Implemented Level 8

ryanjansen committed:
Merge branch 'branch-Level-7'

ryanjansen committed:
Merge branch 'branch-Level-8'

ryanjansen committed:
Implemented A-MoreOOP

ryanjansen committed:
Implemented A-Packages

ryanjansen committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

ryanjansen committed:
Implemented Gradle

ryanjansen committed:
Implemented A-JUnit

ryanjansen committed:
Added java docs

ryanjansen committed:
followed coding standards

ryanjansen committed:
Implemented Level 9

ryanjansen committed:
Merge branch 'branch-A-JavaDoc'

ryanjansen committed:
Merge branch 'branch-A-CodingStandard'

ryanjansen committed:
Merged branch-Level-9 into master

ryanjansen committed:
Added GUI

ryanjansen committed:
Merge branch 'branch-Level-10'

UnwilledTangent commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>(100);
```
Oh? Is there a need to specify the capacity to 100? Why not use the default capacity, which is 
`this.tasks = new ArrayList<>();`

UnwilledTangent commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,9 @@
+package duke.command;
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
```
Might want to put a whitespace like so :)

`package duke.command;`
&ensp;
`import duke.TaskList;`
`import duke.Ui;`
`import duke.Storage;`

UnwilledTangent commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,23 @@
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.task.Task;
+public class DeleteCommand extends Command {
```
Might want to add a whitespace here! :)

UnwilledTangent commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+    public DeleteCommand(int taskIndex) {
+        this.taskIndex = taskIndex;
+    }
+    @Override
```
Whitespace here too!

UnwilledTangent commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,29 @@
+import duke.Ui;
+import duke.Storage;
+import duke.task.Task;
+import java.time.LocalDate;
+public class EventCommand extends Command {
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,29 @@
+        this.title = title;
+        this.from = from;
+        this.to = to;
+    }
+    @Override
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/ExitCommand.java:
```diff
@@ -0,0 +1,17 @@
+
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+public class ExitCommand extends Command{
```
```suggestion
import duke.Storage;

public class ExitCommand extends Command{
```

UnwilledTangent replied:
Whitespace here too! :)

UnwilledTangent replied:
I'm trying the commit feature! Tell me if it works!

UnwilledTangent commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+    public FindCommand(String query) {
+        this.query = query;
+    }
+    @Override
```
```suggestion
    }
    
    @Override
```

Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/ListCommand.java:
```diff
@@ -0,0 +1,16 @@
+
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+public class ListCommand extends Command {
```
```suggestion
import duke.Storage;

public class ListCommand extends Command {
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/TickCommand.java:
```diff
@@ -0,0 +1,23 @@
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.task.Task;
+public class TickCommand extends Command{
```
```suggestion
import duke.task.Task;

public class TickCommand extends Command{
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/TickCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+    public TickCommand(int taskIndex) {
+        this.taskIndex = taskIndex;
+    }
+    @Override
```
```suggestion
    }
    
    @Override
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,23 @@
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.task.Task;
+public class TodoCommand extends Command {
```
```suggestion
import duke.task.Task;

public class TodoCommand extends Command {
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/TodoCommand.java:
```diff
@@ -0,0 +1,23 @@
+
+    public TodoCommand(String title) {
+        this.title = title;
+    }
+    @Override
```
```suggestion
    }
    
    @Override
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/command/UntickCommand.java:
```diff
@@ -0,0 +1,24 @@
+
+    public UntickCommand(int taskIndex) {
+        this.taskIndex = taskIndex;
+    }
+    @Override
```
```suggestion
    }
    
    @Override
```
Whitespace here too! :)

UnwilledTangent commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    /**
+     * Mark the task as done.
```
```suggestion
     * Marks the task as done.
```
According to the coding guideline, should be marks, not mark :)

UnwilledTangent commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    /**
+     * Mark the task as not done.
```
```suggestion
     * Marks the task as not done.
```
same here!

UnwilledTangent commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    /**
+     * Return the display string representing the task.
```
```suggestion
     * Returns the display string representing the task.
```

UnwilledTangent commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    /**
+     * Return the saved string representing the task.
```
```suggestion
     * Returns the saved string representing the task.
```

UnwilledTangent commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,19 @@
+    public Todo(String title) {
+        super(title);
+    }
+
+
```
```suggestion

```
2 spaces here!

UnwilledTangent commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,19 @@
+    public String toString() {
+        return "[T]" + super.toString();
+    }
+
+
```
```suggestion

```
2 spaces here! should be 1!

UnwilledTangent commented on src/test/java/duke/command/ExitCommandTest.java:
```diff
@@ -0,0 +1,13 @@
+import duke.task.Todo;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+public class ExitCommandTest {
```
```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;

public class ExitCommandTest {
```
Whitespace here too! :)

UnwilledTangent commented on text-ui-test/input.txt:
```diff
@@ -0,0 +1,10 @@
+deadline finish ip /by 2359
+event lecture /from now /to later
+list
+todo survive one more day
+mark 1
+list
+mark 3
+unmark 3
+list
+bye
```
No need to push test files! Can gitignore! :)

UnwilledTangent commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -5,3 +5,54 @@ Hello from
+       [E][ ] lecture (from: now to: later)
+     Now you have 2 tasks in the list
+    ____________________________________________________________
+    ____________________________________________________________
+     Here are the tasks in your list:
+     1.[D][ ] finish ip (by: 2359)
+     2.[E][ ] lecture (from: now to: later)
+    ____________________________________________________________
+    ____________________________________________________________
+     Got it. I've added this task:
+       [T][ ] survive one more day
+     Now you have 3 tasks in the list
+    ____________________________________________________________
+    ____________________________________________________________
+     Nice! I've marked this task as done:
+       [D][X] finish ip (by: 2359)
+    ____________________________________________________________
+    ____________________________________________________________
+     Here are the tasks in your list:
+     1.[D][X] finish ip (by: 2359)
+     2.[E][ ] lecture (from: now to: later)
+     3.[T][ ] survive one more day
+    ____________________________________________________________
+    ____________________________________________________________
+     Nice! I've marked this task as done:
+       [T][X] survive one more day
+    ____________________________________________________________
+    ____________________________________________________________
+     OK! I've marked this task as not done yet:
+       [T][ ] survive one more day
+    ____________________________________________________________
+    ____________________________________________________________
+     Here are the tasks in your list:
+     1.[D][X] finish ip (by: 2359)
+     2.[E][ ] lecture (from: now to: later)
+     3.[T][ ] survive one more day
+    ____________________________________________________________
+    ____________________________________________________________
+     Bye. Hope to see you again soon!
+    ____________________________________________________________
```
No need to push test files! Can gitignore! :)

UnwilledTangent reviewed (commented):
I'm very impressed with your coding standards! Well done! The most that I have observed is just whitespaces that need to be added here and there! Other than that, no major violations! Keep up the good work! I'm sure that you will be a good programmer in the tP if you keep this coding standard up! :) Jiayous for the rest of CS2103!

ryanjansen committed:
Improved code quality

Refactored much of the code to make it more readable. Main
changes were removing nested if statements, refactoring them
into switch statements instead and removing unecessary string
builder instances from the UI class.

ryanjansen committed:
Added assertions

Assert that the title of the new tasks added match the
title passed to the respective method.

ryanjansen committed:
Merge pull request #2 from ryanjansen/branch-A-Assertions

Added assertions

ryanjansen committed:
Merge branch 'master' into branch-A-CodeQuality

ryanjansen committed:
Merge pull request #3 from ryanjansen/branch-A-CodeQuality

Improved code quality

ryanjansen committed:
Added snooze extension

ryanjansen committed:
A-UserGuide

ryanjansen committed:
A-Release

ryanjansen committed:
Updated docs
